seed/api/v2_1/views.py

```diff
@@ -241,220 +235,46 @@ def update_with_building_sync(self, request, pk):
 
         the_file = request.data['file']
         file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
-        organization_id = request.data['organization_id']
-        cycle_id = request.data.get('cycle_id', None)
+        organization_id = request.query_params.get('organization_id', None)
+        cycle_pk = request.query_params.get('cycle_id', None)
 
-        if not cycle_id:
+        if not cycle_pk:
             return JsonResponse({
                 'success': False,
                 'message': "Cycle ID is not defined"
             })
         else:
-            cycle = Cycle.objects.get(pk=cycle_id)
+            cycle = Cycle.objects.get(pk=cycle_pk)
 
-        building_file = BuildingFile.objects.create(
-            file=the_file,
-            filename=the_file.name,
-            file_type=file_type,
-        )
+        result = self._get_property_view(pk, cycle_pk)
+        p_status = False
+        new_pv_state = None
+        if result.get('status', None) != 'error':
+            building_file = BuildingFile.objects.create(
+                file=the_file,
+                filename=the_file.name,
+                file_type=file_type,
+            )
 
-        try:
-            property_view = PropertyView.objects.get(pk=pk, cycle=cycle)
-        except PropertyView.DoesNotExist:
-            return JsonResponse({
-                'success': False,
-                'message': 'Cannot match a PropertyView with pk=%s; cycle_id=%s' % (pk, cycle_id)
-            })
+            property_view = result.pop('property_view')
+
+            # passing in the existing property state allows it to process the buildingsync without creating a new state
```

Comment:
OK, so the `process` method appears to now take the old `property_view`.  This is fine, but the comment should be changed.

Reply:
we went back and forth on what to pass... in the end I think this was the best. I'll update the comment.